lightwood/mixer/lightgbm.py

```diff
@@ -182,10 +188,22 @@ def fit(self, train_data: EncodedDs, dev_data: EncodedDs) -> None:
         # Determine time per iterations
         start = time.time()
         self.params['num_iterations'] = 1
-        self.model = lightgbm.train(self.params, lightgbm.Dataset(
-            data['train']['data'],
-            label=data['train']['label_data']),
-            verbose_eval=False)
+        '''
+        Why construst a dataset here instead of using the training dataset?
+
+        Because it guards against the following crash:
+
+        WARNING:lightwood-1613058:Exception: Cannot change feature_pre_filter after constructed Dataset handle. when training mixer: <lightwood.mixer.lightgbm.LightGBM object at 0x7f39148eae20>
+        feature_fraction, val_score: inf:   0%|                                 | 0/7 [00:00<?, ?it/s]
+        free(): double free detected in tcache 2
+
+        Only happens sometimes and I can find no pattern as to when, happens for multiple input and target types.
+
+        Why does the following crash happen and what does it mean? No idea, closest relationships I can find is /w optuna modifying parameters after the dataset is create: https://github.com/microsoft/LightGBM/issues/4019 | But why this would apply here makes no sense. Could have to do with the `train` process of lightgbm itself setting a "set only once" property on a dataset when it starts. Dunno, if you find out replace this comment with the real reason.
```

Comment:
this is really bad if we can't figure out what's wrong

Reply:
It is.... but I've spent some time trying to. Even if we did, likely no fix, when I have more time I will do a self-contained replication and report the bug (maybe contribute a fix) in the LightGBM repo.

Ultimately there's a bunch of "design around bug of our libraries" things in lightwood, including the fact that all encoders are designed using that stupid `is_none` function because pandas can't ****** guarantee that values in a column don't change. It's non ideal, but this is hardly the worst offender.